CICE-interface/CMakeLists.txt

```diff
@@ -9,7 +9,7 @@ elseif(CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
   set(CMAKE_Fortran_FLAGS "-g -traceback")
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -FR -convert big_endian -assume byterecl -ftz -align array64byte -xHOST")
   set(CMAKE_Fortran_FLAGS_RELEASE "-O2 -fp-model precise")
-  set(CMAKE_Fortran_FLAGS_DEBUG "-O0 -check uninit -check bounds -check pointers -fpe0 -check noarg_temp_created -link_mpi=dbg")

```

Comment:
We remove this link_mpi because if it is already specified in Intel.cmake?

Reply:
Yes, I believe that is @aerorahul found. This fix allowed mom6solo to compile on cheyenne. I did test the cpld_debug on hera w/ this change and it did not affect answers. I think this is also the motivation behind issue #330.

Reply:
@junwang-noaa 
We should not set that as `CMAKE_Fortran_FLAGS` as is currently set in `cmake/Intel.cmake`
It  should be part of `CMAKE_Fortran_FLAGS_DEBUG`. 
This is raised in issue #330

Reply:
@aerorahul I am confused about "It should be part of CMAKE_Fortran_FLAGS_DEBUG", the link_mpi was defined in CMAKE_Fortran_FLAGS_DEBUG, but the updated code removes it?

Reply:
It was hard-wired in the `CMake_Fortran_FLAGS_DEBUG` for MOM6 and CICE6 CMakeLists.txt.  This does not work for machines e.g. wcoss2, cheyenne.

That is not the appropriate place for it.  If it should exist, it should be in `Intel.cmake`.